### PR TITLE
chore: update npm linux to v1.5.23 and windows to v1.5.5

### PR DESCRIPTION
**Pull request by rayaisaiah**
**What type of PR is this?** NPM Version Update

**What this PR does / why we need it**: Updates the NPM Linux image version from v1.4.45.3 to v1.5.23 to address Ubuntu repo updates/CVEs, remove building and publishing NPM images with certs/private keys, and aligns NPM's minor version with CNS/CNI versions. Updates NPM Windows image version from v1.4.34 to v1.5.5 to align with image in RP.

**Release note**: https://github.com/Azure/azure-container-networking/releases/tag/v1.5.23

### CODE REVIEW

**Review by huntergregory — APPROVED** [review]

File: vhdbuilder/packer/components.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
       "downloadURL": "mcr.microsoft.com/containernetworking/azure-npm:*",
       "amd64OnlyVersions": [],
       "multiArchVersions": [
-        "v1.4.45.3"
+        "v1.5.23"
```
Comment (jaer-tsun) [inline vhdbuilder/packer/components.json:70]: we usually keep the older version

and any updates on windows side here?

https://github.com/Azure/AgentBaker/blob/4b548c3ec7bb3ceafd67ff3b1c3ee07ee96e1d40/vhdbuilder/packer/generate-windows-vhd-configuration.ps1#L50

Comment (rayaisaiah) [reply vhdbuilder/packer/components.json:70]: Updated the windows versions to align with the versions in RP. I will merge this PR once i confirm theres no need to rollback.

**Review by jaer-tsun — APPROVED** [review]

**Review by AbelHu — CHANGES_REQUESTED** [review]
You need to create a PR from a branch in Azure/AgentBaker instead of the branch in your forked repo to run all tests.

**Comment by rayaisaiah** [comment]
> You need to create a PR from a branch in Azure/AgentBaker instead of the branch in your forked repo to run all tests.

https://github.com/Azure/AgentBaker/pull/4184

**Pull request closed without merging**